compiler-plugin/src/main/kotlin/arrow/meta/phases/analysis/DefaultElementScope.kt

```diff
@@ -334,5 +412,14 @@ class DefaultElementScope(project: Project) : ElementScope {
 
   override val String.annotatedExpression: AnnotatedExpression
     get() = AnnotatedExpression(expression.value as KtAnnotatedExpression)
```

Comment:
@bloderxd @raulraja I've only noticed this is how we're doing a number of string -> psi elements for the ones that are not automatically provided by the `PsiFactory` - why not just do it this way for the remainder of items?

Reply:
yes, we need a matching function in the ElementScope for every Scope we have that turns a template into a scope.

Reply:
Right - but some of the PsiFactory does not have methods available for certain PsiElements that can be used with the `delegate` -  I think what @bloderxd did is an interesting idea, and I wondered if this is a pattern to continue for all elements not easily obtainable from `PsiFactory.kt` or if we should think about changing them all to be similar?

Reply:
Let's think about it and discuss it after KC if you are ok with it.

Reply:
Which method does it not have? Some methods like `.expression`as in the example above account for a big number of element types which result after parsing the structure and are all subtypes of KtExpression.

Reply:
Works for me! Oh, that would take care of many I'm sure. 

I'll have to check for the rest of them, but one example is for `KtCatchClause`